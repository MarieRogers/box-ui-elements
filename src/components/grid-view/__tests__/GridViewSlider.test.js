import React from 'react';

import GridViewSlider from '../GridViewSlider';

const intl = {
    formatMessage: jest.fn(),
};

describe('components/grid-view/GridViewSlider', () => {
    test('should render()', () => {
        const wrapper = mount(
            <GridViewSlider
                columnCount={4}
                maxColumnCountcolumnCount={4}
                gridMaxColumns={7}
                gridMinColumns={2}
                intl={intl}
                maxColumnCount={7}
                onChange={() => {}}
            />,
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('input').prop('aria-label')).toBe('Column size');
        expect(
            wrapper
                .find('PlainButton')
                .at(0)
                .prop('aria-label'),
        ).toBe('Decrease column size');
        expect(
            wrapper
                .find('PlainButton')
                .at(1)
                .prop('aria-label'),
        ).toBe('Increase column size');
    });
});
