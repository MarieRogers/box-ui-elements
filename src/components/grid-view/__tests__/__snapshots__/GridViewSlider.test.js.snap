// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/grid-view/GridViewSlider should render() 1`] = `
<GridViewSlider
  columnCount={4}
  gridMaxColumns={7}
  gridMinColumns={2}
  intl={
    Object {
      "formatMessage": [MockFunction],
    }
  }
  maxColumnCount={7}
  maxColumnCountcolumnCount={4}
  onChange={[Function]}
>
  <GridViewSlider
    columnCount={4}
    gridMaxColumns={7}
    gridMinColumns={2}
    intl={
      Object {
        "formatDate": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatRelativeTime": [Function],
      }
    }
    maxColumnCount={7}
    maxColumnCountcolumnCount={4}
    onChange={[Function]}
  >
    <div
      className="bdl-GridViewSlider"
    >
      <PlainButton
        aria-label="Decrease column size"
        className="bdl-GridViewSlider-button"
        onClick={[Function]}
        type="button"
      >
        <button
          aria-label="Decrease column size"
          className="btn-plain bdl-GridViewSlider-button"
          onClick={[Function]}
          type="button"
        >
          <IconMinusThin
            color="#909090"
            height={14}
            width={14}
          >
            <AccessibleSVG
              className="icon-minus-thin "
              height={14}
              viewBox="0 0 9 1"
              width={14}
            >
              <svg
                aria-hidden="true"
                className="icon-minus-thin "
                focusable="false"
                height={14}
                role="presentation"
                viewBox="0 0 9 1"
                width={14}
              >
                <path
                  className="fill-color"
                  d="M0 0h9v1H0z"
                  fill="#909090"
                  fillRule="evenodd"
                />
              </svg>
            </AccessibleSVG>
          </IconMinusThin>
        </button>
      </PlainButton>
      <input
        aria-label="Column size"
        className="bdl-GridViewSlider-range"
        max={6}
        min={1}
        onChange={[Function]}
        step={1}
        type="range"
        value={3}
      />
      <PlainButton
        aria-label="Increase column size"
        className="bdl-GridViewSlider-button"
        onClick={[Function]}
        type="button"
      >
        <button
          aria-label="Increase column size"
          className="btn-plain bdl-GridViewSlider-button"
          onClick={[Function]}
          type="button"
        >
          <IconPlusThin
            color="#909090"
            height={14}
            width={14}
          >
            <AccessibleSVG
              className="icon-plus-thin "
              height={14}
              viewBox="0 0 9 9"
              width={14}
            >
              <svg
                aria-hidden="true"
                className="icon-plus-thin "
                focusable="false"
                height={14}
                role="presentation"
                viewBox="0 0 9 9"
                width={14}
              >
                <path
                  className="fill-color"
                  d="M5 4V0H4v4H0v1h4v4h1V5h4V4H5z"
                  fill="#909090"
                  fillRule="evenodd"
                />
              </svg>
            </AccessibleSVG>
          </IconPlusThin>
        </button>
      </PlainButton>
    </div>
  </GridViewSlider>
</GridViewSlider>
`;
